book/src/deploying.md

```diff
@@ -1,101 +1,102 @@
 # Deploying CSI Driver on Kubernetes
 
-> ## *This page is out-of-date and under active development.*
-
 This page describes to CSI driver developers how to deploy their driver onto a Kubernetes cluster.
 
 ## Overview
-There are three components plus the kubelet that enable CSI drivers to provide storage to Kubernetes. These components are sidecar containers which are responsible for communication with both Kubernetes and the CSI driver, making the appropriate CSI calls for their respectful Kubernetes events.
 
-## Sidecar Containers
-[![sidecar-container](images/sidecar-container.png)](https://docs.google.com/a/greatdanedata.com/drawings/d/1JExJ_98dt0NAsJ7iI0_9loeTn2rbLeEcpOMEvKrF-9w/edit?usp=sharing)
+A CSI driver is typically deployed in Kubernetes as two components:
+a controller component and a per-node component.
 
-Sidecar containers manage Kubernetes events and make the appropriate calls to the CSI driver. These are the _external attacher_, _external provisioner_, _external snapshotter_ and the _driver registrar_.
+### Controller Plugin
 
-### External Attacher
-[external-attacher](https://github.com/kubernetes-csi/external-attacher) is a sidecar container that watches Kubernetes _VolumeAttachment_ objects and triggers CSI _ControllerPublish_ and _ControllerUnpublish_ operations against a driver endpoint. As of this writing, the external attacher does not support leader election and therefore there can be only one running per CSI driver.  For more information please read [_Attaching and Detaching_](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md#attaching-and-detaching).
+The controller component can be deployed as a Deployment or StatefulSet on
+any node in the cluster. It consists of the CSI driver that implements the
+CSI Controller service and one or more
+[sidecar containers](sidecar-containers.md). These controller
+sidecar containers typically interact with Kubernetes objects and make calls
+to the driver's CSI Controller service.
 
-Note, even though this is called the _external attacher_, its function is to call the CSI API calls _ControllerPublish_ and _ControllerUnpublish_. These calls most likely will occur in a node which is _not_ the one that will mount the volume. For this reason, many CSI drivers do not support these calls, instead doing the attach/detach and mount/unmount both in the CSI _NodePublish_ and _NodeUnpublish_ calls done by the kubelet at the node which is supposed to mount.
+It generally does not need direct access to the host and can perform all its
+operations through the Kubernetes API and external control plane services.
+Multiple copies of the controller component can be deployed for HA, however
+it is recommended to use leader election to ensure there is only one active
+controller at a time.
```

Comment:
Might be worthwhile to document how to do leader election/HA for our sidecars at some point?

Reply:
https://github.com/kubernetes-csi/docs/issues/138 to track that work.